### PR TITLE
feat: add solution to task 2

### PR DESCRIPTION
**Pull request by moshenetsb**

### CODE REVIEW

**Review by ANOM0S — COMMENTED** [review]
co do tabeli wydaje mi się, że jest w niej lekka redundancja danych, oraz może kilka niepotrzebnych kolumn które mogą powodować szumy.
Podział git.
wybrana metryka ma jak najbardziej sens, oraz jest fajnie wytłumaczona.
Ładnie przedstawiony baseline, fajna funkcja do przedstawiania danych. Jest mały problem z komentarzem, możliwe, że ponowne odpalenie kodu zmieniło wynik i się trochę nie zgadza.
Dwa modele ładnie zaimplementowane. Dobrze, ze zwróciłeś uwagę o skalowaniu przy SVM.
Dobrze, że pokazałeś, że stały wybór zbiorów może prowadzić do "szczęśliwego trafu" i pokazałeś jak zmieniają się wyniki po kroswalidacji.
Badanie hiperparametrów również w porządku.
Jeszcze na koniec przyczepie się o brak jakichkolwiek wykresów. Potrafią one w ładny sposób pokazać jak wpływają hiperparametry na wyniki.

**Review by morawskae — COMMENTED** [review]
Dobrze, że użyłeś `random_state` przy `train_test_split`, dzięki temu wyniki są powtarzalne.

Podoba mi się, że nie ograniczyłeś się tylko do accuracy. Wybór metryki `Precision` i, co ważniejsze, uzasadnienie, dlaczego akurat ją wybrałeś, jest na plus.

Duży plus za sprawdzenie SVM przed i po skalowaniu. Dobrze pokazuje to, jak duży wpływ skalowanie może mieć na ten model.

Sprawdziłeś wyniki za pomocą cross-validation, zamiast opierać się tylko na jednym podziale danych, to dobrze. 

Po GridSearchu chętnie zobaczyłabym jeszcze jakąś wizualizację wyników dla poszczególnych hiperparametrów. Ułatwiłoby to zobaczenie, jak zmiana parametrów wpływa na model.

**Comment by Marcelele-0** [comment]
Plusy
- użycie Pipeline i ColumnTransformer to bardzo dobry call, świetnie ucina problem wycieku danych.
- sprawdzenie SVM z i bez RobustScaler mega na plus, fajnie obrazuje wrażliwość modeli opartych na dystansach.
- walidacja przez cross_val_score i wszędobylski random_state dają bardzo stabilne i powtarzalne wyniki.
- GridSearch i wnioski o nieliniowości (kernel RBF) mają spory sens i są trafne.

Minusy
- rozjazd tekstu z wynikiem przy Dummy Classifier, odpalone jeszcze raz bez updatu komentarza.
- optymalizacja pod samo Precision to pułapka (często tnie Recall), bezpieczniej było uderzać w F1-score - tutaj wyszlo okej (jesli to byla przemyslana decyzja to na plus)
 
Ogólnie pipeline napisany bardzo solidnie i z głową, więc mimo braku wykresów i małych potknięć przy feature engineeringu, spokojnie wystarczy, żeby puścić to dalej.

**Review by Marcelele-0 — APPROVED** [review]
